preprocessing/nextclade/src/loculus_preprocessing/embl.py

```diff
@@ -0,0 +1,210 @@
+        gene_qualifiers = getattr(config, "annotations", {}).get("gene_qualifiers", [])
+        gene_attributes_map = {qualifier: qualifier for qualifier in gene_qualifiers}
+        gene_attributes_map.update(attribute_map)
+        gene_range = gene.get("range")
+        attributes = gene.get("attributes", {})
+        qualifiers = {
+            new_key: attributes[old_key]
+            for old_key, new_key in gene_attributes_map.items()
+            if old_key in attributes
+        }
+        qualifiers["codon_start"] = 1
+        # In FeatureLocation start and end are zero based, exclusive end.
+        # thus an embl entry of 123..150 (one based counting) becomes a location of [122:150]
+        feature = SeqFeature(
+            FeatureLocation(start=gene_range["begin"], end=gene_range["end"]),
+            type="gene",
+            qualifiers=qualifiers,
+        )
+        feature_list.append(feature)
+        for cds in gene.get("cdses", []):
+            cds_qualifiers = getattr(config, "annotations", {}).get("cds_qualifiers", [])
+            cds_attributes_map = {qualifier: qualifier for qualifier in cds_qualifiers}
+            cds_attributes_map.update(attribute_map)
+            segments = cds.get("segments", [])
+            ranges = [segment.get("range") for segment in segments]
+            attributes_cds = cds.get("attributes", {})
+            strands = [-1 if segment.get("strand") == "-" else +1 for segment in segments]
+            locations = [
+                FeatureLocation(start=r["begin"], end=r["end"], strand=s)
+                for r, s in zip(ranges, strands, strict=False)
+            ]
+            compound_location = locations[0] if len(locations) == 1 else CompoundLocation(locations)
+            qualifiers = {
+                new_key: attributes_cds[old_key]
+                for old_key, new_key in cds_attributes_map.items()
+                if old_key in attributes_cds
+            }
+            qualifiers["codon_start"] = 1
+            qualifiers["translation"] = "".join(
+                [
+                    str(Seq(sequence_str[(range["begin"]) : (range["end"])]).translate())
+                    for range in ranges
+                ]
+            )
+            feature = SeqFeature(
+                location=compound_location,
+                type="CDS",
+                qualifiers=qualifiers,
+            )
+            feature_list.append(feature)
+    return feature_list
+
+
+def create_flatfile(  # noqa: PLR0913, PLR0917
+    config: Config,
+    accession,
+    version,
+    metadata,
+    unaligned_nucleotide_sequences,
+    annotation_object: dict[str, Any] | None = None,
+) -> str:
+    collection_date = metadata.get("sampleCollectionDate", "Unknown")
+    country = get_country(metadata)
+    description = get_description(accession, version, config.db_name)
+    organism = config.scientific_name
+    authors = get_authors(metadata.get("authors", ""))
```

Comment:
These fields are all created using fields in the Loculus metadata - I think it would be good to add them to the config to make this clear (as if we ever change the name of these fields the embl flat file creation will fail and now that we are doing this in prepro we really want to prevent any errors)

Reply:
Ah yeah. So you mean add a config like "sample_collection_data_field_name = sampleCollectionDate" ?

Reply:
I implemented that here: https://github.com/loculus-project/loculus/pull/4344/commits/320aa86bb1ca33dab4bfbf0c4c32c31c88e21471

Reply:
Thank you!

Reply:
maybe we could make it a bit clearer that we use these fields for the embl flat file, I wonder if we could make a subsection embl_flat_file_fields and then give this dictioanry of values

Reply:
Ah yes that sounds like a good idea